nemo/collections/asr/contextnet.py

```diff
@@ -0,0 +1,275 @@
+# Copyright (c) 2019 NVIDIA Corporation
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+import nemo
+from .parts.jasper import JasperBlock, init_weights, jasper_activations
+from nemo.backends.pytorch.nm import TrainableNM
+from nemo.core.neural_types import *
+from nemo.utils.decorators import add_port_docs
+
+logging = nemo.logging
+
+
+class ContextNetEncoder(TrainableNM):
```

Comment:
Should this inherit from JasperEncoder ?

Reply:
Good point

Reply:
On second thought, it probably should not inherit JasperEncoder. While yes currently they share exactly same functionality, in the future they might not. In that case, the `__init__` call will instantiate multiple JasperBlocks before ContextNetEncoder starts to instantiate its own values.

While there is duplication for now, it is cleaner to separate the two modules

Reply:
Fixed